sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/eventhubs/kafka/AzureEventHubsKafkaAutoConfigurationTests.java

```diff
@@ -0,0 +1,258 @@
+        this.contextRunner
+                .withUserConfiguration(ServiceBusStaticConnectionStringProviderConfiguration.class)
+                .run(context -> {
+                    assertThat(context).hasSingleBean(AzureEventHubsKafkaAutoConfiguration.class);
+                    assertThat(context).doesNotHaveBean(KafkaPropertiesBeanPostProcessor.class);
+                });
+    }
+
+    @Test
+    void shouldConfigureWhenArmConnectionStringProvided() {
+        this.contextRunner
+                .withUserConfiguration(ArmConnectionStringProviderConfiguration.class)
+                .run(context -> {
+                    assertThat(context).hasSingleBean(AzureEventHubsKafkaAutoConfiguration.class);
+                    assertThat(context).hasSingleBean(KafkaPropertiesBeanPostProcessor.class);
+                    assertThat(context).hasSingleBean(KafkaProperties.class);
+
+                    KafkaProperties bean = context.getBean(KafkaProperties.class);
+                    Map<String, String> properties = bean.getProperties();
+                    assertThat(bean.getBootstrapServers()).isEqualTo(Collections.singletonList("arm-namespace.servicebus.windows.net:9093"));
+                    assertThat(properties.get(SECURITY_PROTOCOL_CONFIG)).isEqualTo(SASL_SSL.name());
+                    assertThat(properties.get(SASL_MECHANISM)).isEqualTo("PLAIN");
+                });
+    }
+
+    @Test
+    void shouldConfigureWhenAnotherKafkaPropertiesProvided() {
+        KafkaProperties defaultProperties = new KafkaProperties();
+        defaultProperties.setBootstrapServers(Collections.singletonList("default-namespace.servicebus.windows.net:9093"));
+
+        this.contextRunner
+                .withBean(KafkaProperties.class, () -> defaultProperties, beanDefinition -> beanDefinition.setPrimary(true))
+                .withUserConfiguration(StaticConnectionStringProviderConfiguration.class)
+                .run(context -> {
+                    assertThat(context).hasSingleBean(AzureEventHubsKafkaAutoConfiguration.class);
+                    assertThat(context).hasSingleBean(KafkaPropertiesBeanPostProcessor.class);
+                    Map<String, KafkaProperties> kafkaPropertiesMap = context.getBeansOfType(KafkaProperties.class);
+                    assertThat(kafkaPropertiesMap).hasSize(2);
+
+                    KafkaProperties bean = context.getBean(KafkaProperties.class);
+                    Map<String, String> properties = bean.getProperties();
+                    assertThat(bean.getBootstrapServers()).isEqualTo(Collections.singletonList("static-namespace.servicebus.windows.net:9093"));
+                    assertThat(properties.get(SECURITY_PROTOCOL_CONFIG)).isEqualTo(SASL_SSL.name());
+                    assertThat(properties.get(SASL_MECHANISM)).isEqualTo("PLAIN");
+                });
+    }
+
+    @Test
+    void shouldNotOverrideKafkaPropertiesWithConnectionStringAutoConfiguration() {
```

Comment:
What is this case used for?

Reply:
To make sure the oauth auto configuration won't influence the legacy auto configuration for eh cs.

Reply:
then maybe we should refactor the test case name, since I can't tell the purpose from this name